packages/edit-post/src/components/sidebar/post-template/form.js

```diff
@@ -25,29 +25,26 @@ export default function PostTemplateForm( { onClose } ) {
 		canCreate,
 		canEdit,
 	} = useSelect( ( select ) => {
+		const { canUser, getEntityRecord, getEntityRecords } =
+			select( coreStore );
 		const editorSettings = select( editorStore ).getEditorSettings();
-		const siteSettings = select( coreStore ).getEntityRecord(
-			'root',
-			'site'
-		);
+		const siteSettings = canUser( 'read', 'settings' )
+			? getEntityRecord( 'root', 'site' )
+			: undefined;
```

Comment:
The settings cannot be marked with a view or public context. Related Core PR: https://github.com/WordPress/wordpress-develop/pull/2985